### PR TITLE
specify --profile for cli

### LINKED ISSUE #777: cfn CLI needs a way to specify --profile
We use a lot of profiles (multiple gov profiles, seller profiles, commercial profiles etc.)

Since cfn command can only use default profile, we have to manually edit credentials file to switch between profiles.
This makes it hard to automate testing and publishing. This also makes it hard to test multiple things

Ability to specify profile will be very helpful. AWS Cli also provides following argument
       --profile (string)       Use a specific profile from your credential file.

### PR DESCRIPTION
**Pull request by rrhodes**
closes #777

*Description of changes:*
New optional argument `--profile` to indicate which AWS profile should be used by the CLI for commands `submit`, `invoke`, and `test`.

*Testing:*
In addition to unit test modifications, the following commands have been manually tested with and without `--profile` specified:

- [x] `submit` module
- [x] `submit` hook
- [x] `submit` resource

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ammokhov** [comment]
Thanks for the contribution, we should take a look at your pull-request soon and get back

**Review by Brianwithay21 — APPROVED** [review]
Changes for `HOOK` type specific changes look good to me

**Review by MalikAtalla-AWS — APPROVED** [review]

**Comment by rrhodes** [comment]
Hi @Brianwithay21, @MalikAtalla-AWS, anything further required before this is eligible for merging? I don't appear to have permission to merge this.

**Comment by MalikAtalla-AWS** [comment]
Hey @rrhodes, thanks for the reminder. I'll ping the team who owns this repo.

**Comment by rrhodes** [comment]
> Hey @rrhodes, thanks for the reminder. I'll ping the team who owns this repo.

No trouble, thanks @MalikAtalla-AWS 🙂

File: doc_source/resource-type-cli-invoke.md
Comment (ammokhov) [inline doc_source/resource-type-cli-invoke.md:37]: Would be useful to show how profile should be specified

Comment (rrhodes) [reply doc_source/resource-type-cli-invoke.md:37]: Would you mind elaborating on this, please? We have the `--profile <value>` example above on line 34, consistent with other documented options.

**Review by ericzbeard — CHANGES_REQUESTED** [review]
There are some conflicts with the latest changes on trunk that need to be resolved before we can merge this.

**Review by ericzbeard — APPROVED** [review]

**Pull request merged**
